client/webserver/api.go

```diff
@@ -73,6 +73,47 @@ func (s *WebServer) apiEstimateRegistrationTxFee(w http.ResponseWriter, r *http.
 	writeJSON(w, resp, s.indent)
 }
 
+// apiValidateAddress is the handlers for the '/validateaddress' API request.
+func (s *WebServer) apiValidateAddress(w http.ResponseWriter, r *http.Request) {
+	form := &struct {
+		Addr    string  `json:"addr"`
+		AssetID *uint32 `json:"assetID"`
+	}{}
+	if !readPost(w, r, form) {
+		return
+	}
+	valid := s.core.ValidateAddress(form.Addr, *form.AssetID)
```

Comment:
Can I have you add a `if form.AssetID == nil` error in `apiValidateAddress`, in `apiEstimateSendTxFee`, and also in `apiEstimateRegistrationTxFee` while you're at it?